modular_skyrat/modules/sec_haul/code/melee/baton.dm

```diff
@@ -0,0 +1,556 @@
+// VERY IMPORTANT TO NOTE: Armor with batons is averaged across all limbs, meaning
+// A helmet of melee 2 won't be as effective as a jumpsuit with melee 1.
+
+#define TRANSLATE_EXTRA_SWING_ARMOR(armor) (1 - armor)
+
+/// The default armor penetration of a baton in overcharge mode. Percentage-based.
+#define BATON_OVERCHARGE_ARMOR_PENETRATION_PERCENT 0.85 // keep armor SOMEWHAT effective, at least 15%
+/// The default armor penetration of a baton in overcharge mode. Flat. Needed to be above 1 to get the armor penetration message.
+#define BATON_OVERCHARGE_ARMOR_PENETRATION_FLAT 0.1 // todo: make it so we only need the percentage
+/// The volume of the extra sound played on overcharged baton impact, for extra "oomph" factor
+#define BATON_OVERCHARGE_EXTRA_HITSOUND_VOLUME 20
+
+/// How long it takes for a user to replace a cell in a baton.
+#define BATON_CELL_REPLACE_TIME 3 SECONDS
+
+// Police baton - very hard to not get knocked down by it, though its low attack rate limits its effectiveness
+/obj/item/melee/baton
+	/// The armor flag used when we use our stun function, AKA our left click.
+	var/stun_armor_flag = MELEE
```

Comment:
Needs to be set to energy for stunbatons and kept melee for regular truncheon-like batons like the telescopic baton.